keps/sig-auth/0014-dynamic-audit-configuration.md

```diff
@@ -87,20 +87,65 @@ A cluster scoped configuration object will be provided that applies to all event
 
 ```golang
 // AuditConfiguration represents a dynamic audit configuration
-type AuditConfiguration struct {
+type AuditSink struct {
     metav1.TypeMeta
 
     v1.ObjectMeta
 
-    // Policy is the current audit v1beta1 Policy object
-    // if undefined it will default to the statically configured cluster policy if available
+    // Policy is scoped down version of the V1 Policy object
+    // if undefined it will default to the statically configured cluster policy if available	
     // if neither exist the backend will fail
     Policy *Policy
 
     // Backend to send events
     Backend *Backend
 }
 
+// Level defines the amount of information logged during auditing
+type Level string
+
+// Valid audit levels
+const (
+    // LevelNone disables auditing
+    LevelNone Level = "None"
+    // LevelMetadata provides the basic level of auditing.
+    LevelMetadata Level = "Metadata"
+    // LevelRequest provides Metadata level of auditing, and additionally
+    // logs the request object (does not apply for non-resource requests).
+    LevelRequest Level = "Request"
+    // LevelRequestResponse provides Request level of auditing, and additionally
+    // logs the response object (does not apply for non-resource requests and watches).
+    LevelRequestResponse Level = "RequestResponse"
+)
+
+// Stage defines the stages in request handling during which audit events may be generated.
+type Stage string
+
+// Valid audit stages.
+const (
+    // The stage for events generated after the audit handler receives the request, but before it
+    // is delegated down the handler chain.
+    StageRequestReceived = "RequestReceived"
+    // The stage for events generated after the response headers are sent, but before the response body
+    // is sent. This stage is only generated for long-running requests (e.g. watch).
+    StageResponseStarted = "ResponseStarted"
+    // The stage for events generated after the response body has been completed, and no more bytes
+    // will be sent.
+    StageResponseComplete = "ResponseComplete"
+    // The stage for events generated when a panic occurred.
+    StagePanic = "Panic"
+)
+
+// Policy is a scoped down policy for audit webhooks
+type Policy struct {
+  	// The Level that all requests are recorded at.
+    Level Level
+
+    // OmitStages is a list of stages for which no events are created.
+    // +optional
+    OmitStages []Stage
```

Comment:
Can we please make this a whitelist? I know it deviates from the static version, but IMO not making that a whitelist was a mistake (`IncludeStages`). Must be non-empty (or default to all?).